crowbar_framework/app/models/provisioner_service.rb

```diff
@@ -279,16 +267,25 @@ def disable_repository(platform, arch, repo, bag = nil)
     repo_id = repo_object.id
     @logger.debug("ID for #{repo} is #{repo_id}") if repo_id != repo
 
-    if bag.fetch(platform, {}).fetch(arch, {}).key? repo_id
+    repo_in_db = repo_object.data_bag_item
+    if !repo_in_db.nil?
       @logger.debug("Setting #{repo_id} repository for #{platform} / #{arch} as inactive.")
-      bag[platform][arch].delete(repo_id)
-      bag[platform].delete(arch) if bag[platform][arch].empty?
-      bag.delete(platform) if bag[platform].empty?
-      bag.save
+      repo_in_db.destroy(repo_object.data_bag_name, repo_object.data_bag_item_name)
+      db = repo_object.data_bag
+      if !db.nil? && db.empty?
+        chef_data_bag_destroy(repo_object.data_bag_name)
+      end
     else
       @logger.debug("#{repo_id} repository for #{platform} / #{arch} is already inactive.")
     end
 
     [200, ""]
   end
+
+  private
+
+  # workaround for Chef::DataBag.destroy not working
+  def chef_data_bag_destroy(name)
```

Comment:
Interesting, that doesn't work for me. Can you try the patch and use the proper destroy method to see if that works for you?